source/simulator/helper_functions.cc

```diff
@@ -1320,6 +1323,106 @@ namespace aspect
+  {
+    // Replace FormulationMassConservation::ask_material_model by the respective terms to avoid
+    // complicated checks later on
+    if (parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::ask_material_model)
+      {
+        if (material_model->is_compressible() == true)
+          parameters.formulation_mass_conservation = Parameters<dim>::FormulationMassConservation::isothermal_compression;
+        else
+          parameters.formulation_mass_conservation = Parameters<dim>::FormulationMassConservation::incompressible;
+      }
+
+    // Ensure the material model supports the selected formulation of the mass conservation equation
+    if (parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::incompressible)
+      {
+        AssertThrow(material_model->is_compressible() == false,
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The mass conservation equation was selected to be incompressible, "
+                               "but the provided material model reports to be compressible. "
+                               "Please check the consistency of your material model and selected formulation."));
+      }
+    else if (parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::isothermal_compression
+             || parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::reference_density_profile
+             || parameters.formulation_mass_conservation == Parameters<dim>::FormulationMassConservation::implicit_reference_density_profile)
+      {
+        AssertThrow(material_model->is_compressible() == true,
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The mass conservation equation was selected to be compressible, "
+                               "but the provided material model reports to be incompressible. "
+                               "Please check the consistency of your material model and selected formulation."));
+      }
+
+    // Ensure that the correct heating terms have been selected for the chosen combined formulation
+    // Note that if the combined formulation is 'custom' there is no check
+    // (useful e.g. for smaller scale lithospheric models with shear heating but without adiabatic heating)
+    if (parameters.combined_formulation == Parameters<dim>::CombinedFormulation::isothermal_compression)
+      {
+        AssertThrow(heating_model_manager.adiabatic_heating_enabled(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The 'isothermal compression' formulation expects adiabatic heating to be enabled, "
+                               "but the heating model manager does not find a heating plugin "
```

Comment:
does -> did

Alternatively, say that the 'adiabatic heating' plugin has not been selected in the input file, or similar.